src/middlewared/middlewared/plugins/jail.py

```diff
@@ -715,39 +718,33 @@ def exec(self, jail, command, options):
         host_user = "" if jail_user and host_user == "root" else host_user
         msg = iocage.exec(command, host_user, jail_user, msg_return=True)
 
-        return msg.decode("utf-8")
+        return msg
 
     @accepts(Str("jail"))
     @job(lock=lambda args: f"jail_update:{args[-1]}")
     def update_to_latest_patch(self, job, jail):
         """Updates specified jail to latest patch level."""
+        job.set_progress(0, f'Updating {jail}')
 
-        uuid, path, _ = self.check_jail_existence(jail)
-        status, jid = IOCList.list_get_jid(uuid)
-        conf = IOCJson(path).json_load()
-
-        # Sometimes if they don't have an existing patch level, this
-        # becomes 11.1 instead of 11.1-RELEASE
-        _release = conf["release"].rsplit("-", 1)[0]
-        release = _release if "-RELEASE" in _release else conf["release"]
-
-        started = False
-
-        if conf["type"] == "jail":
-            if not status:
-                self.start(jail)
-                started = True
-        else:
-            return False
-
-        if conf["basejail"] != "yes":
-            IOCFetch(release).fetch_update(True, uuid)
-        else:
-            # Basejails only need their base RELEASE updated
-            IOCFetch(release).fetch_update()
+        def progress_callback(content, exception):
+            msg = content['message'].strip('\n')
 
-        if started:
-            self.stop(jail)
+            if 'Inspecting system... done' in msg:
+                job.set_progress(20, msg)
+            elif 'Preparing to download files... done.' in msg:
+                job.set_progress(50, msg)
+            elif 'Applying patches... done.' in msg:
```

Comment:
Is the output always the same? I remember freebsd used to added more outputm e.g.
Applying patches... 100...200..300... done.

Is that no longer the case?